### ISSUE TITLE
missing TS error when a required query param is not passed

### ISSUE THREAD

**Issue by astahmer** [ecyrbe/zodios#189]
idk if that is even possible but I expected this error at compile-time

```ts
const apiEndpoints = makeApi([
  {
    method: 'delete',
    path: '/cart/clean',
    requestFormat: 'json',
    parameters: [
      {
        name: 'storeId',
        type: 'Query',
        schema: z.number(),
      },
    ],
    response: z.string(),
  },
]);
const client = new Zodios(apiEndpoints);

// no error, should have error
client.delete('/cart/clean');

// no error, should have error
client.delete('/cart/clean', undefined);

// expects error, is fine
client.delete('/cart/clean', undefined, {});

// expects error, is fine
client.delete('/cart/clean', undefined, { queries: {} });
```

I think there is the same behaviour with `header` params

**Comment by astahmer** [comment]
oh ok my bad

**Comment by ecyrbe** [comment]
@astahmer your last two cases however should give you an error. i guess it's because of your typescript config.
can you check that you tried it with a tsconfig with 'strict' enabled ?
if it's not it, can you check your ts version ?

**Comment by astahmer** [comment]
<img width="1057" alt="Screenshot 2022-10-18 at 21 46 11" src="https://user-images.githubusercontent.com/47224540/196529050-ea108c22-ba8e-4d2b-8ea4-025d743f987a.png">

tsconfig.json
```json
{
  "compilerOptions": {
    "target": "ESNext",
    "lib": ["DOM", "DOM.Iterable", "ESNext"],
    "types": ["vite/client", "node"],
    "allowJs": true, // TODO
    "allowSyntheticDefaultImports": true,
    "noImplicitAny": false,
    "module": "ESNext",
    "moduleResolution": "Node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "sourceMap": true,
    "inlineSources": true,
    "jsx": "react-jsx",
    "baseUrl": "./src",
    "paths": {
      "@/*": ["*"]
    },

    // adapted from https://github.com/tsconfig/bases/blob/main/bases/strictest.json
    "strict": true,
    "allowUnusedLabels": false,
    "allowUnreachableCode": false,
    "strictNullChecks": true,
    "exactOptionalPropertyTypes": false,
    "noFallthroughCasesInSwitch": true,
    "noImplicitOverride": true,
    "noImplicitReturns": false,
    "noPropertyAccessFromIndexSignature": false,
    "noUncheckedIndexedAccess": true,
    "noUnusedLocals": true,
    "noUnusedParameters": false, // TODO maybe enable later ?

    "importsNotUsedAsValues": "error",
    "checkJs": false, // TODO maybe enable later ?

    "esModuleInterop": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true
  },
  "include": ["./src"],
  "exclude": [
    "node_modules",
    "dist",
    "cypress",
    "**/*.test.ts",
    "**/*.test.tsx",
    "**/*.test.js",
    "**/*.test.jsx"
  ],
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Strictest"
}
```

> also, can you check the file was not created on an exclude path ?
meaning not in a path included in tsconfig ? I confirm that the screen above was done in a file included in `./src` which is included in the tsconfig

**Comment by ecyrbe** [comment]
Thanks, 
so the last two are giving correct error. I think you only need strict to be true.
You can check zodios project tsconfig, it's really simple.

**Comment by ecyrbe** [comment]
Are your files JS one's ? As i see allowjs to be true...
  
Because in JavaScript only, zodios will only give you autocompletion, not errors unless you add `// @ts-check`

**Comment by ecyrbe** [comment]
fixed in `v10.2.0` : 
- `@zodios/core`
- `@zodios/plugins`
- `@zodios/react`
- `@zodios/solid`
